lib/api/core/plugins/pluginContext.js

```diff
@@ -70,24 +69,67 @@ class PluginContext {
       Dsl: 'Koncorde'
     });
 
-    this.errors = require('kuzzle-common-objects').errors;
-
+    this.errors = errors;
+    this.errorsManager = require('../../../config/error-codes/throw').wrap('plugins', pluginName);
+    
     this.secrets = JSON.parse(JSON.stringify(kuzzle.vault.secrets));
 
     if (kuzzle instanceof Kuzzle) {
       const
-        // Lowercasing the plugin name is needed because Elasticsearch
-        // forbids uppercased characters in index names.
-        internalEngineIndex = `%plugin:${pluginName}`.toLowerCase(),
-        pluginInternalEngine =
-          new InternalEngine(kuzzle, internalEngineIndex),
-        kuzzleSdk = new KuzzleSDK(new FunnelProtocol(kuzzle.funnel));
-
-      pluginInternalEngine.init(
-        new PluginInternalEngineBootstrap(
-          pluginName,
+        kuzzleSdk = new KuzzleSDK(new FunnelProtocol(kuzzle.funnel)),
+        // uppercase are forbidden by ES
+        pluginIndex = `plugin-${pluginName}`.toLowerCase(),
+        pluginIndexStorage = new IndexStorage(
+          pluginIndex,
+          kuzzle.storageEngine.internal);
+
+      Object.defineProperty(this.accessors, 'storage', {
+        enumerable: true,
+        get: () => {
+          return {
+            bootstrap: collections => pluginIndexStorage.init(collections),
+            createCollection: (...args) => pluginIndexStorage.createCollection(...args)
+          };
+        }
+      });
+
+      /**
+       * @todo Use a CollectionEngine class instead repository
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Complete the task associated to this TODO comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS1135)